modules/internal/ChapelArray.chpl

```diff
@@ -1078,6 +1078,37 @@ module ChapelArray {
       for i in _value.dimIter(d, ind) do yield i;
     }
 
+   /* Returns a tuple of integers describing the size of each dimension.
+      For a sparse domain, returns the shape of the parent domain.*/
+    proc shape where isRectangularDom(this) {
+      var shape: rank*(int);
```

Comment:
As a rule, I wouldn't personally declare a variable with the same name as the function it's enclosed in. Nonetheless if it works that's just a style issue.